src/services/refactors/convertToNamedParameters.ts

```diff
@@ -0,0 +1,519 @@
+        Debug.assert(actionName === refactorName);
+        const { file, startPosition, program, cancellationToken, host } = context;
+        const functionDeclaration = getFunctionDeclarationAtPosition(file, startPosition, program.getTypeChecker());
+        if (!functionDeclaration || !cancellationToken) return undefined;
+
+        const groupedReferences = getGroupedReferences(functionDeclaration, program, cancellationToken);
+        if (groupedReferences.valid) {
+            const edits = textChanges.ChangeTracker.with(context, t => doChange(file, program, host, t, functionDeclaration, groupedReferences));
+            return { renameFilename: undefined, renameLocation: undefined, edits };
+        }
+
+        return { edits: [] }; // TODO: GH#30113
+    }
+
+    function doChange(
+        sourceFile: SourceFile,
+        program: Program,
+        host: LanguageServiceHost,
+        changes: textChanges.ChangeTracker,
+        functionDeclaration: ValidFunctionDeclaration,
+        groupedReferences: GroupedReferences): void {
+        const newParamDeclaration = map(createNewParameters(functionDeclaration, program, host), param => getSynthesizedDeepClone(param));
+        changes.replaceNodeRangeWithNodes(
+            sourceFile,
+            first(functionDeclaration.parameters),
+            last(functionDeclaration.parameters),
+            newParamDeclaration,
+            {   joiner: ", ",
+                // indentation is set to 0 because otherwise the object parameter will be indented if there is a `this` parameter
+                indentation: 0,
+                leadingTriviaOption: textChanges.LeadingTriviaOption.IncludeAll,
+                trailingTriviaOption: textChanges.TrailingTriviaOption.Include
+            });
+
+        const functionCalls = sortAndDeduplicate(groupedReferences.functionCalls, /*comparer*/ (a, b) => compareValues(a.pos, b.pos));
+        for (const call of functionCalls) {
+            if (call.arguments && call.arguments.length) {
+                const newArgument = getSynthesizedDeepClone(createNewArgument(functionDeclaration, call.arguments), /*includeTrivia*/ true);
+                changes.replaceNodeRange(
+                    getSourceFileOfNode(call),
+                    first(call.arguments),
+                    last(call.arguments),
+                    newArgument,
+                    { leadingTriviaOption: textChanges.LeadingTriviaOption.IncludeAll, trailingTriviaOption: textChanges.TrailingTriviaOption.Include });
+            }
+        }
+    }
+
+    function getGroupedReferences(functionDeclaration: ValidFunctionDeclaration, program: Program, cancellationToken: CancellationToken): GroupedReferences {
+        const functionNames = getFunctionNames(functionDeclaration);
+        const classNames = isConstructorDeclaration(functionDeclaration) ? getClassNames(functionDeclaration) : [];
+        const names = deduplicate([...functionNames, ...classNames], equateValues);

```

Comment:
I'm not sure I understand why there would be duplicates.

Reply:
when refactoring the constructor of a class expression, the variable name to which the class is assigned is both a function name and a class name.